src/algorithms/eigsolvers/Arnoldi.m

```diff
@@ -25,6 +25,53 @@
         end
         
         function varargout = eigsolve(alg, A, v0, howmany, sigma)
+            % Find a few eigenvalues and eigenvectors of an operator using an Arnoldi
+            % routine.
+            %
+            % Usage
+            % -----
+            % :code:`[V, D, flag] = eigsolve(A, v, howmany, sigma)`
+            % :code:`D = eigsolve(A, v, ...)`
+            %
+            % Arguments
+            % ---------
+            % A : matrix or function_handle
+            %   A square matrix.
+            %   A function handle which implements one of the following, depending on sigma:
+            %
+            %   - A \ x, if `sigma` is 0 or 'smallestabs'
+            %   - (A - sigma * I) \ x, if sigma is a nonzero scalar
+            %   - A * x, for all other cases
+            %
+            % v : vector
+            %   initial guess for the eigenvector.
+            %
+            % howmany : int
+            %   amount of eigenvalues and eigenvectors that should be computed. By default
+            %   this is 1, and this should not be larger than the total dimension of A.
+            %
+            % sigma : `char` or numeric
+            %   selector for the eigenvalues, should be either one of the following:
+            %
+            %   - 'largestabs', 'lm': default, eigenvalues of largest magnitude
+            %   - 'largestreal', 'lr': eigenvalues with largest real part
+            %   - 'largestimag', 'li': eigenvalues with largest imaginary part.
+            %   - 'smallestabs', 'sm': default, eigenvalues of smallest magnitude
+            %   - 'smallestreal', 'sr': eigenvalues with smallest real part
+            %   - 'smallestimag', 'si': eigenvalues with smallest imaginary part.
+            %
+            % Returns
+            % -------
+            % V : (1, howmany) array
+            %   vector of eigenvectors.
+            %
+            % D : numeric
+            %   vector of eigenvalues if only a single output argument is asked, diagonal
+            %   matrix of eigenvalues otherwise.
+            %
+            % flag : int
+            %   if flag = 0 then all eigenvalues are converged, otherwise not.
```

Comment:
other flag values are missing, maybe mention that warnings are suppressed if flag is returned